v7/_posts/odata features/2017-12-26-06-00-JsonBatch.md

```diff
@@ -0,0 +1,330 @@
+                        }
+
+                        // Query various attributes for this operation request such as request dependencies and atomic group associated
+                        // and combine with specific processing of the service.
+                        Foo_ProcessingForExecutionOrders(operationMessage.DependsOnIds);
+                        Foo_ProcessingForAtomicTransaction(operationMessage.GroupId);
+                        // ...
+                        
+                        // Response processing begins: including response writing (depends on whether it is associated with an atomic group) etc. to 
+                        // write back to the HTTP response belonged to implementation of specific services.
+
+                        response.StatusCode = 201;
+                        response.SetHeader("Content-Type", "application/json;odata.metadata=none");
+                        ...
+                        // Response processing ends.
+                    }
+                    else if (operationMessage.Method == "PATCH")
+                    {
+                        // Process flow of PATCH method similar to PUT method above.
+                        // ...
+                    }
+                    // additional else if blocks for other HTTP method processs flow come here...
+
+                break;
+
+                case ODataBatchReaderState.ChangesetStart:
+
+                    // Request's GroupId can be passed to response writer for correlation of groupId between request & response.												
+                    batchWriter.WriteStartChangeset(batchReader.CurrentGroupId);
+
+                    // Handle atomic group starting here as needed by implementation of specific services.
+                    ...
+                    // End of atomic group start processing.
+                    break;
+
+                case ODataBatchReaderState.ChangesetEnd:
+
+                    batchWriter.WriteEndChangeset();
+
+                    // Handle atomic group ending here as needed by implementation of specific services.
+                    ...
+                    // End of atomic group end processing.
+                    break;
+            }
+        }
+        
+        batchWriter.WriteEndBatch();
+{% endhighlight %}
+
+With the introduction of JSON batch, the following public APIs / attributes are available:
+* `ODataBatchWriter`
+    * `public ODataBatchOperationRequestMessage CreateOperationRequestMessage(string method, System.Uri uri, string contentId, BatchPayloadUriOption payloadUriOption, IEnumerable<string> dependsOnIds)`: This can be used to create operation request message with the additional dependsOnIds that explicitly specify operation dependencies in a JSON batch request. For Multipart/Mixed batch, this new overloaded API can also be used, as well as the two original APIs without the need to explicitly specify the dependsOnIds, which will be derived implicitly per the protocol.
```

Comment:
Sentence is a bit awkward.  Maybe "This new overload can also be used for Multipart/Mixed to validate the dependencies. Calling an overload without dependencyIDs in Multipart/Mixed validates any referenced statement results are valid according to the protocol.

Reply:
Yep that is better. Updated. Thanks.